extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,145 @@
+import { utils } from 'mocha';
+import * as vscode from 'vscode';
+import { AutoLispExt } from '../extension';
+import { Sexpression } from '../format/sexpression';
+import { ReadonlyDocument } from '../project/readOnlyDocument';
+import { escapeRegExp } from "../utils";
+
+
+export class AutolispDefinitionProvider implements vscode.DefinitionProvider{
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
+		const editor: vscode.TextEditor = vscode.window.activeTextEditor;
+		const rDoc = ReadonlyDocument.getMemoryDocument(document);
+		let selected: string =  editor.document.getText(editor.selection);
+		if (selected === '') {
+			rDoc.atomsForest.forEach(sexp => {
+				if (sexp instanceof Sexpression && sexp.contains(position)){
+					const atom = sexp.getAtomFromPos(position);
+					if (atom) {
+						selected = atom.symbol.replace(/^[']*/, '');
+					}
+				}
+			});
+		}
+		if (['(', ')', '\'', '.'].includes(selected)){
```

Comment:
maybe add` ;`

Reply:
Thank you, I will certainly add this to avoid unnecessary work.

On some level, I wish AutoLisp had more restrictive naming so we could just use \W to exclude all singular symbols, but sadly `(defun # () (princ "yo!"))` is a perfectly valid function definition. I actually use that one for a custom string interpolation function actually.

Reply:
Maybe add to check if the current atom is in comment, avoid to do work.